.github/workflows/lint.yml

```diff
@@ -0,0 +1,169 @@
+# ============================================================================ #
+#
+# Run `clang-tidy' over the codebase and report warnings as annotations.
+#
+# ---------------------------------------------------------------------------- #
+
+name: 'Lint'
+on:
+  # No need to run on `main` since we have `pull_request'.
+  workflow_dispatch:
+    inputs:
+      all:
+        description: 'Check all files even if they are unchanged'
+        required: false
+        type: boolean
+  pull_request:
+    paths:
+      - '.github/workflows/lint.yml'
+      - 'pkgdb/src/**'
+      - 'pkgdb/include/**'
+      - 'pkgdb/test/**'
+
+concurrency:
+  group: "${{ github.workflow }}-${{ github.head_ref || github.sha }}"
+  cancel-in-progress: true
+
+jobs:
+  lint-clang-tidy:
+    name: 'Clang Tidy'
+    runs-on: 'ubuntu-latest'
+
+    steps:
+      - name: 'Checkout'
+        uses: 'actions/checkout@v4'
+
+      # Create a list of target files.
+      # If the workflow was triggered by a PR, only check changed files.
+      # If the workflow was triggered explicitly then we _might_ be
+      # explicitly asked to check all files using an input param.
+      - name: 'Create Target List ( PR )'
+        if: ${{ github.event.pull_request }}
+        run: |
+          git fetch origin main;
+          echo "Checking changed sources" >&2;
+          ./pkgdb/build-aux/changed-sources.sh > targets;
+
+      - name: 'Create Target List ( Dispatch )'
```

Comment:
Question: What is the `( Dispatch )` supposed to mean here?

Reply:
it's a workflow trigger category